pkg/secrethub/client.go

```diff
@@ -290,3 +291,34 @@ func (c *Client) userAgent() string {
 
 	return userAgent
 }
+
+// Resolve fetches the value of a secret, when the `ref` parameter has the
+// format `secrethub://<path>`. Otherwise it returns `ref` unchanged, as an array of bytes.
+func (c *Client) Resolve(ref string) ([]byte, error) {
```

Comment:
Consider moving these functions to their own file: env.go. You could even make a separate package called env (which would make env.Resolve and env.ResolveAll) or make a separate 'subservice' called client.Env().Resolve(), or placing these functions on the client.Secrets() subservice.

All in all, these functions are more helper functions that use the client than that they are core client functionality. They cannot exist without a client, so it makes sense to add them, but we can structure them a bit more logically.

Reply:
I moved them in the secretService
They can be called in the following way:
`client.Secrets().Resolve(ref)`
`client.Secrets().ResolveEnv(os.Environ())`